src/index.tsx

```diff
@@ -137,15 +139,24 @@ export default class InfiniteScroll extends Component<Props, State> {
     }
   }
 
-  UNSAFE_componentWillReceiveProps(props: Props) {
+  componentDidUpdate(prevProps: Props) {
     // do nothing when dataLength is unchanged
-    if (this.props.dataLength === props.dataLength) return;
+    if (this.props.dataLength === prevProps.dataLength) return;
 
     this.actionTriggered = false;
-    // update state when new data was sent in
-    this.setState({
-      showLoader: false,
-    });
```

Comment:
If you remove this the loader will never be hidden. Were you able to test this in local?